dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -120,6 +107,11 @@
         return components.response();
     }
 
+    private void onCompletion() {
+        numRunningRequests.decrementAndGet();
```

Comment:
just to make sure I'm understanding this stuff correctly - the book-keeping we do to update numRunningRequests is purely for metrics purposes right? it doesn't actually influence any of the behaviour?

Reply:
That's correct, I have a ticket filed for that as well since it's used for a gauge and we have multiple QueuedChannel instances. Replacing the atomiclongs with metric Counter instances should do the trick though.